src/main/java/org/mondemand/Client.java

```diff
@@ -550,7 +553,7 @@ public void increment (StatType type, String key, int value) {
     if(realValue == null) {
       // create the counter if doesn't exist
       realValue = new StatsMessage(realKey, type);
-      this.stats.put(realKey, realValue);
+      this.stats.putIfAbsent(realKey, realValue);
```

Comment:
I feel snapshot is something that is not there yet and the new package is going to remove snapshot.